src/perception/road_signs_classifier/road_signs_classifier/road_signs_classifier.py

```diff
@@ -0,0 +1,76 @@
+"""
+Package: road_signs_classifier
+File: road_signs_classifier.py
+Author: Pranav Boyapati
+"""
+
+import rclpy
+import base64
+import os
+from rclpy.node import Node
+from sensor_msgs.msg import Image
+from cv_bridge import CvBridge
+from navigator_msgs.msg import RoadSigns
+from navigator_msgs.msg import RoadSignsDetection
+import cv2
+from inference_sdk import InferenceHTTPClient
+
+
+class road_signs_classifier(Node):
+    def __init__(self):
+        super().__init__('road_signs_classifier')
+
+        #instantiate model client
+        self.CLIENT = InferenceHTTPClient(api_url="http://localhost:6060", api_key="BmqYjCBXZD1iPIyq09sG")
+
+        #create subscriptions
+        self.camera_sub = self.create_subscription(Image, '/cameras/camera0', self.image_callback, 10)
+
+        #create variables to store subscription info
+        self.bridge = CvBridge()
+
+        #create publisher
+        self.road_signs_publisher = self.create_publisher(RoadSignsDetection, '/road_signs/detections', 10)
+
+
+    #callbacks for subscriptions
+    def image_callback(self, msg : Image):
+        cv_image = self.bridge.imgmsg_to_cv2(msg, desired_encoding="passthrough") 
+        cv2.imwrite("image.jpeg", cv_image) 
+        self.classify_sign()
+
+    
+    #main control function
+    def classify_sign(self):
+        #gets prediction
+        if (os.path.exists("image.jpeg")):
+            with open("image.jpeg", "rb") as img_file:
```

Comment:
Is there any reason why we write the image to disk and read it back instead of just passing it as a parameter?

Reply:
It was causing all sorts of errors when I tried to pass the image as a parameter, but those all disappeared when writing to disk and reading it back, so figured that was the best way to go about doing this

Reply:
Have you tried something like: 

```
_, jpeg_image = cv2.imencode('.jpg', cv_image)
base64.b64encode(jpeg_image).decode('utf-8')
```

Reply:
I just tried that and it works. Previously when I tried `cv2.imencode`, I did not do `b64encode` because the articles I had read said it is unnecessary in that case, but putting it in now made it work. I guess those articles were wrong